### ISSUE TITLE
Create table properties does not support boolean value

### ISSUE THREAD

**Issue by tlegrave** [apache/iceberg-python#895]
### Apache Iceberg version

main (development)

### Please describe the bug 🐞

Hello there,

When creating a table, I can pass properties like bloom filters like so:

```py
catalog.create_table(
    "default.example",
    schema=Schema(
        NestedField(1, "created_at", TimestampType(), required=True)
    ),
    properties={"write.parquet.bloom-filter-enabled.column.material_id": "true"},
)
```

This gives me the correct configuration for the table:

```sql
CREATE TABLE iceberg.default.example (
   created_at timestamp(6) NOT NULL
)
WITH (
   format = 'PARQUET',
   format_version = 2,
   location = 's3://test-bucket/default/example',
   parquet_bloom_filter_columns = ARRAY['material_id']
)
```

However, if I use a boolean instead of a string in the properties (e.g. `{"write.parquet.bloom-filter-enabled.column.material_id": True}`, it silently ignores the filter.

```sql
CREATE TABLE iceberg.default.example (
   created_at timestamp(6) NOT NULL
)
WITH (
   format = 'PARQUET',
   format_version = 2,
   location = 's3://test-bucket/default/example'
)
```

**Comment by syun64** [comment]
I believe the TableProperties are expected to be a dictionary of string key-value pairs. But the silent swallowing of the table property is definitely troublesome.

**Comment by syun64** [comment]
@tlegrave thank you for raising this. From what I can see, this PR introduced the ability to handle cast non-string inputs in the TableProperties dict when setting the Table Properties: https://github.com/apache/iceberg-python/pull/504

I'm having a difficult time reproducing issue because I'm seeing that the property is being set as: `metadata.properties.get("write.parquet.bloom-filter-enabled.column.material_id") == "True"`, which is expected.

On a tangent, I see that this property is still [unsupported in PyIceberg for writes](https://github.com/apache/iceberg-python/blob/77a07c90b7ca05c5d915c5c02047807c76b5031e/pyiceberg/io/pyarrow.py#L2054-L2064) - just wanted to make note of that before you dived in deeper into trying to enable this flag for the table

**Comment by tlegrave** [comment]
Thank you for your reply @syun64.

> On a tangent, I see that this property is still [unsupported in PyIceberg for writes](https://github.com/apache/iceberg-python/blob/77a07c90b7ca05c5d915c5c02047807c76b5031e/pyiceberg/io/pyarrow.py#L2054-L2064) - just wanted to make note of that before you dived in deeper into trying to enable this flag for the table

Yes, I'm aware of this limitation, and for now we only use pyiceberg to create tables. Write operations are done with spark to handle this bloom filter feature and to bypass some other known limitations of pyiceberg.

Regarding the issue, I made some further tests based on what you mentionned.

This code also ignores the bloom filter (although passing directly "True" as string). Would it be a case sensitive issue?

```py
catalog.create_table(
    "default.example",
    schema=Schema(
        NestedField(1, "created_at", TimestampType(), required=True)
    ),
    properties={"write.parquet.bloom-filter-enabled.column.material_id": "True"},
)
```

For info, I'm extracting the schema after creation of the table doing a `SHOW create table catalog.default.example` on Trino.

Thanks

**Comment by kevinjqliu** [comment]
> This code also ignores the bloom filter (although passing directly "True" as string). Would it be a case sensitive issue?

Likely, can you set it to lowercase and see? 
In the Trino repo, all the references are lowercased

https://github.com/search?q=repo%3Atrinodb%2Ftrino%20write.parquet.bloom-filter-enabled&type=code

**Comment by kevinjqliu** [comment]
Ah, if you pass in `True` (boolean value), this function will turn it into `"True"`
https://github.com/kevinjqliu/iceberg-python/blob/428b894aacb107547ba41433b4b6f37e1ad1914a/pyiceberg/types.py#L65-L70

which is kind of a footgun

```
>>> str(True)
'True'
```

**Comment by kevinjqliu** [comment]
Created #919 to track the fix

**Comment by kevinjqliu** [comment]
Resolved in #924
